### PR TITLE
Cache render_for_web field for alertgroups list serializer

### PR DESCRIPTION
**Pull request by iskhakov**
# What this PR does
This PR caches the field `render_for_web` with lifetime 1 day and cache becomes invalid if it was created before 
* last alert received
* template changed

## Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated

### CODE REVIEW

**Comment by Konstantinov-Innokentii** [comment]
Some time ago we had a discussion about storing first render result for alert-group and not to re-render on templates changes, what do you think?

**Comment by iskhakov** [comment]
Let's try caching to fix draining SLO, let's do that iteration after

File: engine/apps/api/serializers/alert_group.py [hunk 1]
```diff
@@ -92,10 +95,33 @@ def get_render_for_web(self, obj):
         if not obj.last_alert:
             return {}
 
-        return AlertGroupWebRenderer(obj, obj.last_alert).render()
+        web_templates_modified_at = obj.channel.web_templates_modified_at
```
Comment (vadimkerr) [inline engine/apps/api/serializers/alert_group.py:98]: Shouldn't this be on `AlertGroupListSerializer.get_render_for_web` as well so the list view is cached? I believe `ShortAlertGroupSerializer` is only used to render the root alert group and dependant alert groups.
Probably this caching stuff can be moved to a separate function that's used both on `AlertGroupListSerializer` and `ShortAlertGroupSerializer`?

Comment (iskhakov) [reply engine/apps/api/serializers/alert_group.py:98]: yep, thank you, I want to use this approach in all such serializer methods after I confirm it is working on dev cluster

File: engine/apps/api/serializers/alert_group.py [hunk 2]
```diff
@@ -92,10 +95,33 @@ def get_render_for_web(self, obj):
...
+        CACHE_LIFEIME = 60 * 60 * 24
+        cached_render_for_web = cache.get(CACHE_KEY, None)
+
+        # use cache only if cache exists
+        # and cache was created after the last alert created
+        # and either web templates never modified
+        # or cache was created after templates were modified
+        if (
+            cached_render_for_web is not None
+            and cached_render_for_web.get("cache_created_at") > last_alert_created_at
```
Comment (Konstantinov-Innokentii) [inline engine/apps/api/serializers/alert_group.py:111]: Is cache_created_at value from cache datetime so you can compare it with last_alert_created_at? I remember, that after putting to and retrieving from cache and became a string.

File: engine/apps/api/serializers/alert_group.py (hunk 1 above)
Comment (vadimkerr) [reply engine/apps/api/serializers/alert_group.py:98]: Ignore this comment, it's already on `AlertGroupListSerializer`, it's just GitHub merging diff lines 🙂

**Review by vadimkerr — APPROVED** [review]

**Pull request merged**